junit-platform-console/src/main/java/org/junit/platform/console/tasks/ConsoleTestExecutor.java

```diff
@@ -101,10 +104,16 @@ private TestExecutionSummary executeTests(PrintWriter out, Optional<Path> report
 		Launcher launcher = launcherSupplier.get();
 		SummaryGeneratingListener summaryListener = registerListeners(out, reportsDir, launcher);
 
-		LauncherDiscoveryRequestBuilder discoveryRequestBuilder = toDiscoveryRequestBuilder(discoveryOptions);
-		reportsDir.ifPresent(dir -> discoveryRequestBuilder.configurationParameter(OUTPUT_DIR_PROPERTY_NAME,
-			dir.toAbsolutePath().toString()));
-		launcher.execute(discoveryRequestBuilder.build());
+		PrintStream originalOut = System.out;
+		PrintStream originalErr = System.err;
+		try {
+			redirectStdStreams(outputOptions.getStdoutPath(), outputOptions.getStderrPath());
```

Comment:
The `PrintStreams` created in this method should be closed after resetting `System.out`/`System.err` to ensure any buffers are flushed to the output file(s).